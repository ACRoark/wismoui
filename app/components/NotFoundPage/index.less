@import (reference) '../../styles/themes/index.less';

@not-found-page-cls-prefix: ~'@{dte-prefix}-not-found-page';

.@{not-found-page-cls-prefix} {
  color: @dte-gray-600;
  margin: 110px 0px;

  .button {
    display: block;
    margin: 30px auto 0px 0px;
  }

  .divider {
    display: table-cell;
    vertical-align: middle;

    > hr {
      border: solid @dte-gray-300 1px;
      transform: rotate(90deg);
      width: 40px;
    }
  }

  .image {
    float: right;
  }

  .message {
    width: 100%;
  }

  .subtitle {
    display: table-cell;
    font-size: @dte-wismo-font-size-5;
    padding-left: 19px;
    vertical-align: middle;
  }

  .title {
    display: table-cell;
    font-size: @dte-wismo-font-size-6;
    font-weight: bold;
    padding-right: 19px;
    vertical-align: middle;
  }

  .titleRow {
    display: table;
    height: 40px;
    margin-bottom: 30px;
  }

  @media @mobile {
    margin: 60px 0px;

    .button {
      display: block;
      margin: 90px auto;
    }

    .divider {
      display: block;
      text-align: center;

      > hr {
        border: solid @dte-gray-300 1px;
        transform: rotate(0deg);
        width: 100%;
      }
    }

    .image {
      text-align: center;
      float: none;

      > svg {
        height: 86px;
      }
    }

    .message {
      text-align: center;
    }

    .subtitle {
      display: block;
      font-size: @dte-wismo-font-size-5;
      padding-left: 0px;
      padding-top: 12px;
      text-align: center;
    }

    .title {
      display: block;
      font-size: @dte-wismo-font-size-6;
      font-weight: bold;
      padding: 12px 0px;
      text-align: center;
    }

    .titleRow {
      margin-bottom: 24px;
      width: 100%;
    }
  }
}
