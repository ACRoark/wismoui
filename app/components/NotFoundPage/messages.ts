import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.not-found-page';

export default defineMessages({
  backButtonText: {
    id: `${scope}.back-button-text`,
  },
  message: {
    id: `${scope}.message`,
  },
  subtitle: {
    id: `${scope}.subtitle`,
  },
  title: {
    id: `${scope}.title`,
  },
});
