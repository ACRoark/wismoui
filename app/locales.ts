export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
