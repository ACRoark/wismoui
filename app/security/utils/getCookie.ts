const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
