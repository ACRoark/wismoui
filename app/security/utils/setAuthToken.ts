import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
