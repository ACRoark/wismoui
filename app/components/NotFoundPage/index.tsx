import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import ColumnLayout from 'components/ColumnLayout';

import messages from './messages';

import './index.less';

const NotFoundPage: React.FC = (): React.ReactElement => {
  const onButtonClick = (): void => {
    window.location.href = document.referrer;
  };

  return (
    <ColumnLayout>
      <div className="dte-wismo-not-found-page">
        <div className="image">
          <svg width="137" height="181" viewBox="0 0 137 181" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path
              d="M76.7802 167.987C82.8022 175.52 92.5879 180.04 103.126 180.04C121.945 180.04 137 164.974 137 146.141C137 132.582 129.473 121.282 118.181 116.009V43.6917L81.2966 0H12.7967C6.02196 0 0 5.27322 0 12.8063V155.934C0 162.714 5.26916 168.74 12.0439 168.74L76.7802 167.987ZM72.2637 161.96H12.7967C9.78569 161.96 6.77468 158.947 6.77468 155.934V12.8063C6.77468 9.79305 9.78569 6.77981 12.7967 6.77981H75.2747V37.6652C75.2747 44.445 80.5439 50.4715 87.3186 50.4715H112.159V113.749C109.148 112.996 106.137 112.242 103.126 112.242C84.3077 112.242 69.2527 127.309 69.2527 146.141C68.5 151.414 70.0054 156.687 72.2637 161.96ZM81.2966 9.79298L110.654 43.6917H87.3186C83.5549 43.6917 81.2966 40.6784 81.2966 37.6652V9.79298ZM103.126 174.013C87.3187 174.013 75.2747 161.207 75.2747 146.141C75.2747 131.075 88.0714 118.269 103.126 118.269C118.181 118.269 130.978 131.075 130.978 146.141C130.978 161.961 118.181 174.013 103.126 174.013ZM81.2966 143.128V149.154H124.956V143.128H81.2966Z"
              fill="#D9D9D9"
            />
          </svg>
        </div>
        <div className="titleRow">
          <div className="title">
            <FormattedMessage {...messages.title} />
          </div>
          <div className="divider">
            <hr />
          </div>
          <div className="subtitle">
            <FormattedMessage {...messages.subtitle} />
          </div>
        </div>
        <div className="message">
          <FormattedMessage {...messages.message} />
        </div>
        <Button className="button" type="primary" onClick={onButtonClick}>
          <FormattedMessage {...messages.backButtonText} />
        </Button>
      </div>
    </ColumnLayout>
  );
};

export default NotFoundPage;
