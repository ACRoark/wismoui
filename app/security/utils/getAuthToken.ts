import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
