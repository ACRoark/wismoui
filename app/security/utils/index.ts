import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
