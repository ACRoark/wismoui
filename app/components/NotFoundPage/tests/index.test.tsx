import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import NotFoundPage from '..';

describe('<App />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(<Provider store={createMockStore()}><NotFoundPage /></Provider>, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should render correctly when the current language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(<Provider store={createMockStore()}><NotFoundPage /></Provider>, language);

      expect(tree).toMatchSnapshot();
    });
  });
});
