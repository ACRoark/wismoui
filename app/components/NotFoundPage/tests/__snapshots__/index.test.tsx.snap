// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly when the current language is en 1`] = `
<div
  className="dte-wismo-column-layout"
>
  <div
    className="dte-wismo-not-found-page"
  >
    <div
      className="image"
    >
      <svg
        fill="none"
        height="181"
        viewBox="0 0 137 181"
        width="137"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M76.7802 167.987C82.8022 175.52 92.5879 180.04 103.126 180.04C121.945 180.04 137 164.974 137 146.141C137 132.582 129.473 121.282 118.181 116.009V43.6917L81.2966 0H12.7967C6.02196 0 0 5.27322 0 12.8063V155.934C0 162.714 5.26916 168.74 12.0439 168.74L76.7802 167.987ZM72.2637 161.96H12.7967C9.78569 161.96 6.77468 158.947 6.77468 155.934V12.8063C6.77468 9.79305 9.78569 6.77981 12.7967 6.77981H75.2747V37.6652C75.2747 44.445 80.5439 50.4715 87.3186 50.4715H112.159V113.749C109.148 112.996 106.137 112.242 103.126 112.242C84.3077 112.242 69.2527 127.309 69.2527 146.141C68.5 151.414 70.0054 156.687 72.2637 161.96ZM81.2966 9.79298L110.654 43.6917H87.3186C83.5549 43.6917 81.2966 40.6784 81.2966 37.6652V9.79298ZM103.126 174.013C87.3187 174.013 75.2747 161.207 75.2747 146.141C75.2747 131.075 88.0714 118.269 103.126 118.269C118.181 118.269 130.978 131.075 130.978 146.141C130.978 161.961 118.181 174.013 103.126 174.013ZM81.2966 143.128V149.154H124.956V143.128H81.2966Z"
          fill="#D9D9D9"
        />
      </svg>
    </div>
    <div
      className="titleRow"
    >
      <div
        className="title"
      >
        404 ERROR
      </div>
      <div
        className="divider"
      >
        <hr />
      </div>
      <div
        className="subtitle"
      >
        Page not found
      </div>
    </div>
    <div
      className="message"
    >
      The page you requested could not be found.
    </div>
    <button
      className="button dte-wismo-button dte-wismo-button-primary"
      data-track="mimo"
      data-track-action="click"
      data-track-detail="order tracker"
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>
</div>
`;

exports[`<App /> should render correctly when the current language is es 1`] = `
<div
  className="dte-wismo-column-layout"
>
  <div
    className="dte-wismo-not-found-page"
  >
    <div
      className="image"
    >
      <svg
        fill="none"
        height="181"
        viewBox="0 0 137 181"
        width="137"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M76.7802 167.987C82.8022 175.52 92.5879 180.04 103.126 180.04C121.945 180.04 137 164.974 137 146.141C137 132.582 129.473 121.282 118.181 116.009V43.6917L81.2966 0H12.7967C6.02196 0 0 5.27322 0 12.8063V155.934C0 162.714 5.26916 168.74 12.0439 168.74L76.7802 167.987ZM72.2637 161.96H12.7967C9.78569 161.96 6.77468 158.947 6.77468 155.934V12.8063C6.77468 9.79305 9.78569 6.77981 12.7967 6.77981H75.2747V37.6652C75.2747 44.445 80.5439 50.4715 87.3186 50.4715H112.159V113.749C109.148 112.996 106.137 112.242 103.126 112.242C84.3077 112.242 69.2527 127.309 69.2527 146.141C68.5 151.414 70.0054 156.687 72.2637 161.96ZM81.2966 9.79298L110.654 43.6917H87.3186C83.5549 43.6917 81.2966 40.6784 81.2966 37.6652V9.79298ZM103.126 174.013C87.3187 174.013 75.2747 161.207 75.2747 146.141C75.2747 131.075 88.0714 118.269 103.126 118.269C118.181 118.269 130.978 131.075 130.978 146.141C130.978 161.961 118.181 174.013 103.126 174.013ZM81.2966 143.128V149.154H124.956V143.128H81.2966Z"
          fill="#D9D9D9"
        />
      </svg>
    </div>
    <div
      className="titleRow"
    >
      <div
        className="title"
      >
        404 ERROR
      </div>
      <div
        className="divider"
      >
        <hr />
      </div>
      <div
        className="subtitle"
      >
        Página no encontrada
      </div>
    </div>
    <div
      className="message"
    >
      No se pudo encontrar la página que solicitó.
    </div>
    <button
      className="button dte-wismo-button dte-wismo-button-primary"
      data-track="mimo"
      data-track-action="click"
      data-track-detail="order tracker"
      onClick={[Function]}
    >
      Regresa
    </button>
  </div>
</div>
`;
